### PR TITLE
Use custom pygments lexer

### LINKED ISSUE #14623: Update pygments lexer for Chapel
PR #14276 disables syntax highlighting for certain code blocks in the .rst language spec because otherwise the spec does not render due to Pygments lexing errors.

The errors in particular have to do with methods like `proc (atomic int).method() { }`.

We would like to take these steps:

 1. Include the Chapel Pygments lexer for use in chpldoc within the main Chapel repository. That would allow us to update it and keep chpldoc functioning even if updates to the Pygments repository take longer.
 2. Update the Chapel Pygments lexer to handle code like `proc (atomic int).method() { }`.
 3. Remove the workarounds for lexing errors from the .rst spec (e.g. by reverting 534af8633ce5eb97a502b9f38ddf12cf20a107ad )

@ben-albrecht wrote in https://github.com/chapel-lang/chapel/pull/14276#issuecomment-545976007 :

> Unfortunately, there's a pretty large turn-around time for getting changes into the Pygments repository. It would be nice if we could rely on a Chapel pygments lexer that was tied to the chapel repo, similar to what we're doing with sphinxcontrib-chapeldomain. However, unlike the sphinxcontrib, having Chapel in the main pygments repo could be useful to others, so we should continue to promote changes upstream over time.
> 
> Here's a starting point on including a custom lexer in a sphinx project:
> 
> https://stackoverflow.com/questions/16469869/custom-syntax-highlighting-with-sphinx

### PR DESCRIPTION
**Pull request by e-kayrakli**
This PR adds a custom pygments lexer and uses it. Currently this lexer is just a copy of what we have in pygments 2.7.2.

Partially resolves https://github.com/chapel-lang/chapel/issues/14623
Supersedes https://github.com/chapel-lang/chapel/pull/16557

### CODE REVIEW

**Comment by e-kayrakli** [comment]
@lydia-duncan @ben-albrecht -- here's what I have.

This is similar to my initial packaging on the Chapel side: the lexer is added just as a module and not a fully separate package. It felt a bit like an overkill to me.

For testing, I have modified the chpldoc-requirements.txt on the Chapel side to point to this branch of sphinxcontrib. I was able to `make docs` and making changes on the lexer showed on the docs properly (so it actually overrides the pygments' builtin Chapel lexer)

If there's a better way to test it, I'd be happy to do that.

**Comment by e-kayrakli** [comment]
If we were to merge this, I have https://github.com/chapel-lang/chapel/pull/16562 that changes this lexer to resolve the issues we have with the docs today (at least the ones that I am aware of). I'll split that PR into two:

1. change the lexer here
2. change spec to add `.. code-block:: chapel` in some blocks back to highlight them properly

File: sphinxcontrib/chapeldomain/__init__.py [hunk 1]
```diff
@@ -29,8 +29,9 @@
 from sphinx.util.docfields import Field, GroupedField, TypedField
 from sphinx.util.nodes import make_refnode
 
+from sphinxcontrib.chapeldomain.chapel import ChapelLexer
 
-VERSION = '0.0.18'
+VERSION = '0.0.19'
```
Comment (lydia-duncan) [inline sphinxcontrib/chapeldomain/__init__.py]: The version needs to be updated in a separate PR when we've pushed a new release with this in it.  The latest release is 0.0.17, this one should go into 0.0.18

**Review by lydia-duncan — COMMENTED** [review]
Just a minor change

File: sphinxcontrib/chapeldomain/README.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+`chapel.py` is a module that is a local snapshot of the  Pygments highlighter
```
Comment (ben-albrecht) [inline sphinxcontrib/chapeldomain/README.rst]: Does this file get rendered anywhere via Sphinx? If not, might make sense to make it markdown. If so, we should use double tics:

```suggestion
``chapel.py`` is a module that is a local snapshot of the  Pygments highlighter
```

**Review by ben-albrecht — COMMENTED** [review]
An additional minor comment. Looking good overall
